src/net.cpp

```diff
@@ -2864,8 +2864,17 @@ bool CConnman::ForNode(NodeId id, std::function<bool(CNode* pnode)> func)
     return found != nullptr && NodeFullyConnected(found) && func(found);
 }
 
-int64_t PoissonNextSend(int64_t nNow, int average_interval_seconds) {
-    return nNow + (int64_t)(log1p(GetRand(1ULL << 48) * -0.0000000000000035527136788 /* -1/2^48 */) * average_interval_seconds * -1000000.0 + 0.5);
+int64_t CConnman::PoissonNextSendInbound(int64_t now, int average_interval_seconds)
+{
+    if (next_send_inv_to_incoming < now) {
```

Comment:
Given that you're reading from the atomic variable multiple times here, there is a risk that it was updated by two threads in parallel, resulting in two different timestamps being returned from this function - which would break the independence of sends between threads.

This is a highly unlikely event, and it may even be impossible for now (due to limited threading in the network handing code). However, with the objective of doing it "by the book", I believe a correct implementation would be this:

```c++
int64_t previous = next_send_inv_to_incoming.load();
if (previous >= now) return old;
int64_t next = PoissonNextSend(now, average_interval_seconds);
if (next_send_inv_to_incoming.compare_exchange_strong(previous, next)) {
    return next;
} else {
    // Another thread changed next_send_to_inv after we fetched previous, but before
    // we tried to update it. The values chosen by the other thread is loaded in previous,
    // so return that.
    return previous;
}
```